### PR TITLE
User timeline

### PR DESCRIPTION
**Pull request by JACornelius**

### CODE REVIEW

**Comment by lee-benjamin** [comment]
Good practice for dev work on a real team is to rebase your feature branch before you request review.

File: src/js/main.js [hunk 1]
```diff
@@ -41,18 +50,50 @@ class Timeline extends React.Component {
...
+	componentDidMount() {
+		getHomeTimeline(this.homeTimelineCallback);
+		getUserTimeline(this.userTimelineCallback);
+	}
+
+	render(){
+		{this.homeTimelineResult()};
+		{this.userTimelineResult()};
+		return React.createElement('div', {className: 'Timeline'},
+					[React.createElement('div', {className: 'homeTimelineContainer', key: 'homeTimelineCont'},
```
Comment (lee-benjamin) [inline src/js/main.js]: for the future you can define your own function that stands in for React.createElement so your code is less verbose

File: src/js/services.js [hunk 2]
```diff
@@ -43,6 +48,19 @@ const parseJSON = (response) => {
...
+		.then(parseJSON)
+		.then(function(data) {
+			callback(data, false);
+		})
+		.catch(function(error) {
+			callback(null, true);
+		})
+}
+
+
```
Comment (lee-benjamin) [inline src/js/services.js]: new line

File: src/sass/style.scss [hunk 3]
```diff
@@ -16,88 +16,112 @@
...
-				width: 150px;
-				@include text(15px);
-				font-weight: bold;
-			
+.Timeline{
+	width: 1500px;
+	margin: auto;
+	.homeTimelineContainer, .userTimelineContainer {
+	position: relative;
+		#homeTimelinePlaceholder, #userTimelinePlaceholder {
```
Comment (lee-benjamin) [inline src/sass/style.scss]: formatting

File: src/js/components.js [hunk 4]
```diff
@@ -4,14 +4,44 @@ import {getTimeline} from './services'
 
 class TimelineResultComp extends React.Component {
 	render() {
-		return React.createElement('div', {id: 'timelinePlaceholder', className: `${this.props.className}`}, this.props.timelineResult);
+		return React.createElement('div', {id: this.props.timelinePlaceholder, className: `${this.props.className}`, key: 'timelineResComp'}, this.props.timelineResult);
```
Comment (andrewcchau) [inline src/js/components.js]: Is the form for '${this.props.className}' needed?

File: src/js/components.js [hunk 5]
```diff
@@ -4,14 +4,44 @@ import {getTimeline} from './services'
...
 	}
 }
 
-class HomeTimelineButton extends React.Component {
+class TimelineButton extends React.Component {
 	render(){
-		return React.createElement('button', {className: 'timelineButton', key: 'timelineButton', onClick: this.props.onClickFunc}, 'Get Home Timeline');
+		return React.createElement('button', {className: `${this.props.className}`, key: 'timelineButton', onClick: this.props.onClickFunc}, this.props.buttonText);
```
Comment (andrewcchau) [inline src/js/components.js]: Same as above

File: __tests__/TimelineResultCompTest.js [hunk 6]
```diff
@@ -1,21 +1,22 @@
...
-						   "twitterHandle":"JosephineCorn10",
-						   "profileImageUrl":"http://pbs.twimg.com/profile_images/1031635661701308416/C0nXsZv0_normal.jpg",
-						   "statusId":"1035247174618099712",
-						   "createdAt": 1535657135000}];
+
 
 describe("TimelineResultComp", () => {
 	let wrapper;
+    let mockHttpRespText = [{"message":"mackelmorer AND WE DANCEEDDDDDe",
+                           "userName":"Josephine Cornelius",
```
Comment (lee-benjamin) [inline __tests__/TimelineResultCompTest.js]: align the keys

File: __tests__/TimelineCompTest.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+	it("creates correct button according to type", function() {
+		expect(wrapper.find("button").hasClass("userTimelineButton")).toEqual(true);
+	});
+
+	it("creates correct TimelineResultComponent according to type", function() {
+		wrapper.setProps({resultOutput: "testResultOutput"});
+		wrapper.setProps({resultClass: "testResultClass"});
+		expect(wrapper.containsMatchingElement(
+			React.createElement(TimelineResultComp, {timelinePlaceholder: "userTimelinePlaceholder", className: "testResultClass", timelineResult: "testResultOutput"}, null))).toEqual(true);
```
Comment (lee-benjamin) [inline __tests__/TimelineCompTest.js]: use local variables and/or shorthand for React.createElement.

File: __tests__/TimelineTest.js [hunk 8]
```diff
@@ -1,28 +1,29 @@
...
-	let mockHttpRespText = [{"message":"mackelmorer AND WE DANCEEDDDDDe",
-							 "userName":"Josephine Cornelius",
-							 "twitterHandle":"JosephineCorn10",
-							 "profileImageUrl":"http://pbs.twimg.com/profile_images/1031635661701308416/C0nXsZv0_normal.jpg",
-							 "statusId":"1035247174618099712",
-							 "createdAt":1535657135000}];
+	let mockEmptyHttpRespText = [];
+	let mockHttpRespText = {"message":"mackelmorer AND WE DANCEEDDDDDe",
+								 "userName":"Josephine Cornelius",
+								 "twitterHandle":"JosephineCorn10",
```
Comment (lee-benjamin) [inline __tests__/TimelineTest.js]: space after :
align keys

File: __tests__/UserTest.js [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+	let wrapper;
+	let mockTweetObj = {"message":"mackelmorer AND WE DANCEEDDDDDe",
+							 "userName":"Josephine Cornelius",
+							 "twitterHandle":"JosephineCorn10",
+							 "profileImageUrl":"http://pbs.twimg.com/profile_images/1031635661701308416/C0nXsZv0_normal.jpg",
+							 "statusId":"1035247174618099712",
+							 "createdAt":1535657135000};
+
+	beforeEach(function() {
+		wrapper = mount(React.createElement(User, {rawUserTweetObj: mockTweetObj, index: 1}, null));
```
Comment (lee-benjamin) [inline __tests__/UserTest.js]: 4 or 2 spaces?

File: src/js/components.js [hunk 10]
```diff
@@ -4,14 +4,44 @@ import {getTimeline} from './services'
...
+				React.createElement('div', {className: 'twitterHandle', key: 'twitterHandle' + i}, userTweetObj.twitterHandle)
+			]);
+	}
+}
+
+class TimelineComp extends React.Component {
+	render() {
+		if(this.props.timelineType == 'User') {
+			return React.createElement('div', {className: 'userTimelineContainer', key: 'userTimelineCont'},
+					 	[React.createElement('h2', {className: 'header', key: 'userTimelineHeader'}, 'User Timeline'),
```
Comment (lee-benjamin) [inline src/js/components.js]: same

File: src/sass/style.scss [hunk 11]
```diff
@@ -16,88 +16,112 @@
...
+	float: left;
+	.homeTimelineButton {
+		position: absolute;
+		left: 0px;
+		top: 100px
+	}
+	#homeTimelinePlaceholder {
+		.tweet {
+			&:nth-child(odd) {
+					background-color: #e8f5fd;
```
Comment (lee-benjamin) [inline src/sass/style.scss]: spaces

File: src/sass/style.scss [hunk 12]
```diff
@@ -16,88 +16,112 @@
...
-	width: 1000px;
-	margin: 10px auto auto;
-
-	border: {
-		style: solid;
-		width: 1px;
+	#userTimelinePlaceholder {
+		.tweet {
+			&:nth-child(odd) {
+					background-color: #f5f8fa;
```
Comment (lee-benjamin) [inline src/sass/style.scss]: ditto

**Review by lee-benjamin — APPROVED** [review]
fix them and merge

**Review by lee-benjamin — CHANGES_REQUESTED** [review]
just kidding, please disable handle for user timeline.

File: src/sass/style.scss (hunk 12 above)
Comment (lee-benjamin) [reply src/sass/style.scss]: also I neglected to notice this but please don't hardcode the background color... have your tweets share the same css and override the color somehow.

**Review by lee-benjamin — CHANGES_REQUESTED** [review]

**Pull request merged**